app/app.rb

```diff
@@ -70,39 +70,45 @@ def current_user
       redirect "/users/spaces/#{params[:space_id]}"
     else
 
-    start_day = DateTime.parse(params[:available_date_start])
-    finish_day = DateTime.parse(params[:available_date_finish])
-
-    if start_day > finish_day
-      flash.next[:errors] = ['Start date must be before end date']
-      redirect "/users/spaces/#{params[:space_id]}"
-    elsif DateTime.now > start_day
-      flash.next[:errors] = ['That ship has sailed']
-      redirect "/users/spaces/#{params[:space_id]}"
-    else
-
-      days = (start_day..finish_day).group_by(&:day).map { |_,day| day }
-      space  = Space.get(params[:space_id])
-
-      days.each do |available_day|
-        available_date = AvailableDate.first_or_create(date: available_day)
-        unless available_date.spaces.include?(space)
-          available_date.spaces << space
-          available_date.save
+      start_day = DateTime.parse(params[:available_date_start])
+      finish_day = DateTime.parse(params[:available_date_finish])
+
+      if start_day > finish_day
+        flash.next[:errors] = ['Start date must be before end date']
+        redirect "/users/spaces/#{params[:space_id]}"
+      elsif DateTime.now > start_day
+        flash.next[:errors] = ['That ship has sailed']
+        redirect "/users/spaces/#{params[:space_id]}"
+      else
+
+        days = (start_day..finish_day).group_by(&:day).map { |_,day| day }
+        space  = Space.get(params[:space_id])
+
+        days.each do |available_day|
+          available_date = AvailableDate.first_or_create(date: available_day)
+          unless available_date.spaces.include?(space)
+            available_date.spaces << space
+            available_date.save
+          end
         end
+        redirect "/users/spaces/#{params[:space_id]}"
```

Comment:
this needs some serious refactor - is this all one controller? There is at least one class to be extracted from all this.